src/app/RecordingMetadata/BulkEditLabels.tsx

```diff
@@ -246,14 +254,23 @@ export const BulkEditLabels: React.FC<BulkEditLabelsProps> = ({
       ]).subscribe((parts) => {
         const currentTarget = parts[0];
         const event = parts[1];
-        if (currentTarget?.connectUrl != event.message.target && currentTarget?.jvmId != event.message.jvmId) {
-          return;
-        }
-        setRecordings((old) =>
-          old.map((o) =>
-            o.name == event.message.recordingName ? { ...o, metadata: { labels: event.message.metadata.labels } } : o,
-          ),
-        );
+
+    const isMatch = ((currentTarget?.connectUrl === event.message.target) || (currentTarget?.jvmId === event.message.recording.jvmId))
```

Comment:
`event.message.recording.jvmId` - that's funny looking. The server is sending a structure where the `jvmId` is embedded within the `recording`?

Reply:
https://github.com/cryostatio/cryostat3/pull/319/files#r1550302346

Reply:
Yes, this was also one of the reasons labels wont update until after refresh or navigating away.